### PR TITLE
[jwt-verifier] feat: Add verifyIdToken()

### PR DESCRIPTION
**Pull request by denysoblohin-okta**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [guidelines](/okta/okta-oidc-js/blob/master/CONTRIBUTING.md#commit)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Adding Tests
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation changes
- [ ] Other... Please describe:

## What is the current behavior?
Only access tokens can be validated.

Issue Number: N/A

## What is the new behavior?
ID tokens can be validated.

## Does this PR introduce a breaking change?
- [x] Yes
- [ ] No

Verifier will throw the error "No KID specified" if no KID is present in the JWT header.

## Other information
Validation spec: https://github.com/okta/oss-technical-designs/blob/master/technical_designs/jwt-validation-libraries.md#id-token-verification
Internal ref: [OKTA-234446](https://oktainc.atlassian.net/browse/OKTA-234446)

### CODE REVIEW

**Review by oleksandrpravosudko-okta — APPROVED** [review]

File: packages/jwt-verifier/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+# 1.1.0
+
+### Features
+
+- [#951](https://github.com/okta/okta-oidc-js/pull/951) - Add verifyIdToken()
```
Comment (shuowu) [inline packages/jwt-verifier/CHANGELOG.md:5]: nit: Add -> Adds

File: packages/jwt-verifier/README.md [hunk 2]
```diff
@@ -82,7 +106,7 @@ const verifier = new OktaJwtVerifier({
 });
 ```
 
-Validation fails and an error is returned if an access token does not have the configured claim.
+Validation fails and an error is returned if a token does not have the configured claim.
```
Comment (shuowu) [inline packages/jwt-verifier/README.md:109]: nit: a -> the

File: packages/jwt-verifier/lib.js [hunk 3]
```diff
@@ -89,12 +89,36 @@ function verifyAudience(expected, aud) {
...
+}
+
 function verifyIssuer(expected, issuer) {
   if( issuer !== expected ) {
     throw new Error(`issuer ${issuer} does not match expected issuer: ${expected}`);
   }
 }
 
+function verifyNonce(expected, nonce) {
+  if( !!nonce && !expected ) {
```
Comment (shuowu) [inline packages/jwt-verifier/lib.js:111]: nit: `!!` can be removed

**Review by shuowu — APPROVED** [review]

**Review by aarongranick-okta — CHANGES_REQUESTED** [review]
Please bump the version in `jwt-verifier` package.json to `1.1.0`. In this repo we must manage versions manually.

File: packages/jwt-verifier/README.md [hunk 4]
```diff
@@ -3,9 +3,9 @@
 [![npm version](https://img.shields.io/npm/v/@okta/jwt-verifier.svg?style=flat-square)](https://www.npmjs.com/package/@okta/jwt-verifier)
 [![build status](https://img.shields.io/travis/okta/okta-oidc-js/master.svg?style=flat-square)](https://travis-ci.org/okta/okta-oidc-js)
 
-This library verifies Okta access tokens (issued by [Okta Custom Authorization servers](https://developer.okta.com/docs/concepts/auth-servers/) by fetching the public keys from the JWKS endpoint of the authorization server. If the access token is valid it will be converted to a JSON object and returned to your code. 
+This library verifies Okta access tokens (issued by [Okta Custom Authorization servers](https://developer.okta.com/docs/concepts/auth-servers/) and ID tokens by fetching the public keys from the JWKS endpoint of the authorization server. If the access token is valid it will be converted to a JSON object and returned to your code.
```
Comment (aarongranick-okta) [inline packages/jwt-verifier/README.md:6]: This section is talking about id and access tokens together, but there are some important and subtle distinctions between them. the ID token is part of OIDC, (establishing identity) and the access token is part of OAuth (scoping permissions). I think we will want to add more links to the access token section for api access management and how to use scopes. And we will want to add more to id token section about how to use claims within an id token. So I think it would be good to separate the current text into the two sections, "Access Tokens", "ID Tokens"

Access Tokens
This library verifies Okta access tokens .....  
You can learn about access tokens....
> custom auth servers warning...

ID Tokens
> Library verifies ID tokens issued by custom auth servers OR okta org
> How ID tokens are verified by this SDK
> Learn more about ID tokens and OIDC

File: packages/jwt-verifier/lib.js [hunk 5]
```diff
@@ -115,15 +139,19 @@ class OktaJwtVerifier {
     });
     this.verifier = nJwt.createVerifier().setSigningAlgorithm('RS256').withKeyResolver((kid, cb) => {
       this.jwksClient.getSigningKey(kid, (err, key) => {
+        // If kid is undefined, but there is 1 signing key, `jwks-rsa` will not throw error
+        if (!kid)
```
Comment (aarongranick-okta) [inline packages/jwt-verifier/lib.js:143]: is there a JIRA issue related to this undefined kid condition? This seems like its fixing a bug. Should it be in a separate commit?
Also, would it be better to check this condition before calling `getSigningKey` ?

Comment (denysoblohin-okta) [reply packages/jwt-verifier/lib.js:143]: There is [a test for this condition](https://github.com/okta/okta-oidc-js/blob/master/packages/jwt-verifier/test/spec/verifier.spec.js#L100).
But currently, it depends on the configuration of the authorization server. 
If there are 2 public keys in JWKS, `jwks-rsa` will [throw the error](https://github.com/auth0/node-jwks-rsa/blob/4446484a5373d7a65873616a7c4d7889e48f00f3/src/JwksClient.js#L103), which will result in [error from njwt](https://github.com/jwtk/njwt/blob/d6f81191addfdbee5d146bd4a6f3e8b783802b41/index.js#L379):  "Error while resolving signing key for kid  undefined"
If there is 1 key in JWKS, no error will be thrown and tests will fail (which happens currently)

Comment (denysoblohin-okta) [reply packages/jwt-verifier/lib.js]: Maybe remove code change and skip test `should fail if no kid is present in the JWT header`?

Comment (aarongranick-okta) [reply packages/jwt-verifier/lib.js]: If "kid" is required, why don't we throw before calling `getSigningKey`? Is there another condition we are missing here? Should it be

```
if (!kid && !err)
```
or
```
if (!id && key) // this seems to match the comment? (kid is undefined, but there is 1 signing key)
```

The other question is if you think this is a bug in jwk-rsa or njwt? If so we can include the fix here but in a comment reference another issue to fix it in the other library.

**Comment by aarongranick-okta** [comment]
@denysoblohin-okta do you think any of the changes here are "breaking". Should we bump the version to 2.0.0 ?

File: packages/jwt-verifier/lib.js (hunk 5 above)
Comment (denysoblohin-okta) [reply packages/jwt-verifier/lib.js]: I latest commit I've changed behavior to [throw an error if KID is not specified](https://github.com/okta/okta-oidc-js/pull/951/files#diff-944b6965a1656adcd833a6937e8d2e7328110332310c7a607e225b0097b487d8R146) , before calling `getSigningKey `

**Comment by denysoblohin-okta** [comment]
> @denysoblohin-okta do you think any of the changes here are "breaking". Should we bump the version to 2.0.0 ?

@aarongranick-okta I've bumped to 2.0.0 and added "Verifier will throw error "No KID specified" if no KID is present in the JWT header" in chnagelog

**Review by aarongranick-okta — APPROVED** [review]
Looks good!

If possible, could you squash to 2 commits:
- 1 which contains the feature, all code changes and README (add verifyId token)
- 1 which has package.json, CHANGELOG (release 2.0.0)
